packages/editor/src/package/editor.tsx

```diff
@@ -9,18 +10,30 @@ import { SupportedLanguage } from '@editor/types/language-data'
 import { TemplatePluginType } from '@editor/types/template-plugin-type'
 import React from 'react'
 
-import '@/assets-webkit/styles/serlo-tailwind.css'
 import { defaultSerloEditorProps } from './config'
 import { editorData } from './editor-data'
+import { getEditorVersion } from './editor-version'
+import {
+  type StorageFormat,
+  createEmptyDocument,
+  getCurrentDatetime,
+  migrate,
+  type EditorVariant,
+} from './storage-format'
 import { InstanceDataProvider } from '@/contexts/instance-context'
 import { LoggedInDataProvider } from '@/contexts/logged-in-data-context'
+import '@/assets-webkit/styles/serlo-tailwind.css'
 
 export interface SerloEditorProps {
   children: EditorProps['children']
   plugins?: (EditorPluginType | TemplatePluginType)[]
-  initialState?: EditorProps['initialState']
-  onChange?: EditorProps['onChange']
+  initialState?: unknown // Either type `StorageFormat` or outdated storage format that will be migrated to `StorageFormat`
```

Comment:
idea: union type here.
more explicit type and we wouldn't need the comment maybe.

Reply:
We could type it to the current (and all previous) `StorageFormat` variants. But then every integration would need to do the type checking before passing it to the `SerloEditor` component. So, I figured putting the check in the editor package might be easier. What do you think?

Reply:
Sounds reasonable 👍 Thanks for the explanation.